packages/wrangler/src/publish.ts

```diff
@@ -290,14 +311,22 @@ export default async function publish(props: Props): Promise<void> {
   }
 
   const destination = props.outDir ?? (await tmp.dir({ unsafeCleanup: true }));
-  const envName = props.env ?? "production";
+  const envName = props.env ?? DEFAULT_ENV_NAME;
+
+  if (namespaceName && envName != DEFAULT_ENV_NAME) {
+      logger.warn(
+        `Publishing to namespace '${namespaceName}': a non-default environment was specified, but namespaced scripts do not have environments.`
+      );
+  }
 
   const start = Date.now();
   const notProd = Boolean(!props.legacyEnv && props.env);
   const workerName = notProd ? `${scriptName} (${envName})` : scriptName;
-  const workerUrl = notProd
-    ? `/accounts/${accountId}/workers/services/${scriptName}/environments/${envName}`
-    : `/accounts/${accountId}/workers/scripts/${scriptName}`;
+  const workerUrl = namespaceName
+    ? `/accounts/${accountId}/workers/dispatch/namespaces/${namespaceName}/scripts/${scriptName}`
+    : notProd
+      ? `/accounts/${accountId}/workers/services/${scriptName}/environments/${envName}`
+      : `/accounts/${accountId}/workers/scripts/${scriptName}`;
```

Comment:
Maybe this shouldn't be a ternary anymore 😅

Reply:
Maybe a nested `if/else` 
```suggestion
if (const workerUrl = namespaceName) {
	`/accounts/${accountId}/workers/dispatch/namespaces/${namespaceName}/scripts/${scriptName}`
} else {
	if (notProd) {
		`/accounts/${accountId}/workers/services/${scriptName}/environments/${envName}`
	} else {
		`/accounts/${accountId}/workers/scripts/${scriptName}`;
	}
}
```

Reply:
This code doesn't work, if/else doesn't "return" statements. The ternary is fine.

Reply:
I should have had the returns lol sounds good with the ternary just curious what our cutoff on chained ternaries are then?

Reply:
It did seem a bit... 'verbose' to me. I'll try breaking it out into a function.
This is used elsewhere too, I think? I guess the other place I saw it was tests though.